Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -1617,9 +1616,15 @@ namespace AzToolsFramework
         message.Match(AZ::DocumentPropertyEditor::Nodes::Adapter::QueryKey, showKeyQueryDialog);
     }
 
-    void DocumentPropertyEditor::RegisterHandlerPool(AZStd::shared_ptr<AZ::InstancePoolBase> handlerPool)
+    void DocumentPropertyEditor::RegisterHandlerPool(AZ::Name handlerName, AZStd::shared_ptr<AZ::InstancePoolBase> handlerPool)
     {
-        m_handlerPools.push_back(handlerPool);
+        if (m_handlerPools.find(handlerName) != m_handlerPools.end())
+        {
+            AZ_Assert(m_handlerPools[handlerName] == handlerPool, "Registering a handler name for a different handler pool.");
+        }
+
+        // will not overwrite if there is a registered handler pool for the handler name
```

Comment:
Nit: this could mention that it won't overwrite because of the unordered_map, just to make it more clear.